pkg/tests/ossm/smoke_test.go

```diff
@@ -19,47 +19,123 @@ import (
+func TestBasics(t *testing.T) {
+	NewTest(t).Groups(ARM, Full, Smoke, InterOp).Run(func(t TestHelper) {
+		t.Log("Test basics of SMCP: deploy, upgrade, bookinfo and uninstall")
 		ns := "bookinfo"
 
 		t.Cleanup(func() {
 			app.Uninstall(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
 		})
 
-		DeployControlPlane(t)
-
-		t.LogStep("Install bookinfo pods with sidecar and sleep pod without sidecar")
-		app.InstallAndWaitReady(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
-
-		t.LogStep("Check whether sidecar is injected in all bookinfo pods")
-		assertSidecarInjectedInAllBookinfoPods(t, ns)
-
-		t.LogStep("Check if bookinfo productpage is running through the Proxy")
-		oc.Exec(t,
-			pod.MatchingSelector("app=sleep", ns), "sleep",
-			"curl -sI http://productpage:9080",
-			assert.OutputContains(
-				"HTTP/1.1 200 OK",
-				"ProductPage returns 200 OK",
-				"ProductPage didn't return 200 OK"),
-			assert.OutputContains(
-				"server: istio-envoy",
-				"HTTP header 'server: istio-envoy' is present in the response",
-				"HTTP header 'server: istio-envoy' is missing from the response"),
-			assert.OutputContains(
-				"x-envoy-decorator-operation",
-				"HTTP header 'x-envoy-decorator-operation' is present in the response",
-				"HTTP header 'x-envoy-decorator-operation' is missing from the response"))
+		toVersion := env.GetSMCPVersion()
+		fromVersion := toVersion.GetPreviousVersion()
+
+		t.NewSubTest(fmt.Sprintf("install bookinfo with smcp %s", fromVersion)).Run(func(t TestHelper) {
+
+			t.LogStepf("Create SMCP %s and verify it becomes ready", fromVersion)
+			assertSMCPDeploysAndIsReady(t, fromVersion)
+
+			t.LogStep("Install bookinfo pods with sidecar and sleep pod without sidecar")
+			app.InstallAndWaitReady(t, app.Bookinfo(ns), app.SleepNoSidecar(ns))
+			start := time.Now()
+
+			t.LogStep("Check whether sidecar is injected in all bookinfo pods")
+			assertSidecarInjectedInAllBookinfoPods(t, ns)
+
+			t.LogStep("Check if bookinfo productpage is running through the Proxy")
+			retry.UntilSuccess(t, func(t test.TestHelper) {
+				oc.Exec(t,
+					pod.MatchingSelector("app=sleep", ns), "sleep",
+					"curl -sI http://productpage:9080",
+					assert.OutputContains(
+						"HTTP/1.1 200 OK",
+						"ProductPage returns 200 OK",
+						"ProductPage didn't return 200 OK"),
+					assert.OutputContains(
+						"server: istio-envoy",
+						"HTTP header 'server: istio-envoy' is present in the response",
+						"HTTP header 'server: istio-envoy' is missing from the response"),
+					assert.OutputContains(
+						"x-envoy-decorator-operation",
+						"HTTP header 'x-envoy-decorator-operation' is present in the response",
+						"HTTP header 'x-envoy-decorator-operation' is missing from the response"))
+			})
+
+			t.LogStep("Validate that proxy time is less than 5 seconds")
+			elapsed := time.Since(start)
+			t.Logf("Timer stopped: %s", elapsed)
+			if elapsed.Seconds() > 5 {
+				t.Errorf("Proxy took too long to start: %s", elapsed)
+			}
```

Comment:
This doesn't test anything, because you start the timer *after* Bookinfo is ready.  You also shouldn't start the timer earlier, because then you'll also measure the time it takes to pull the bookinfo images, which will cause the test to be flaky.

What you should actually check is how much time passes between the time the proxy container is created/started and the time it becomes ready. You could do that either by inspecting the pod YAML (compare the proxy container's `status.containerStatuses.state.running.startedAt` and `status.conditions[type=Ready].lastTransitionTime`), or by checking the timestamps in the istio-proxy logs (but I'm not sure what line in the logs indicates that the proxy is ready).

Reply:
OK, I'll check this

Reply:
Done